File: be_camphub/src/main/java/org/camphub/be_camphub/dto/response/account/AccountResponse.java
package org.camphub.be_camphub.dto.response.account;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class AccountResponse {
    String id;
    String username;
    String firstname;
    String lastname;
    String email;
    String phoneNumber;
    String idNumber;
    String avatar;

    Double trustScore;
    Double coinBalance;

    String userType;
    String status;

    String createdAt;
    String updatedAt;
}

File: fe_camphub/src/libs/core/dto/response/account.response.ts
import { UserStatus, UserType } from "../../constants";

export interface AccountResponse {
    id: string;
    username: string;
    firstname: string;
    lastname: string;
    email: string;
    phoneNumber: string;
    idNumber: string;
    avatar: string;
    trustScore: number;
    coinBalance: number;
    userType: UserType;
    status: UserStatus;
    createdAt: string; // ISO datetime
    updatedAt: string; // ISO datetime
}

export interface TopUpResponse {
    newBalance: number;
}